src/extensionFunctions.ts

```diff
@@ -292,18 +292,64 @@ export namespace extensionFunctions {
 				}
 			} else if (uri.fsPath) {
 				out = uri;
+			} else {
+				out = vscode.Uri.parse(uri.toString());
 			}
 		}
 		return out;
 	}
 
 	// open the didact window with the didact file passed in via Uri
-	export async function startDidact(uri:vscode.Uri, viewColumn?: vscode.ViewColumn) {
+	export async function startDidact(uri:vscode.Uri, viewColumn?: any) {
 		if (!uri) {
 			uri = await utils.getCurrentFileSelectionPath();
 		}
 
-		sendTextToOutputChannel(`Starting Didact window with ${uri}`);
+		// if column passed as number, convert to viewcolumn enum
```

Comment:
What if the user would like to use ViewColumn.Active or ViewColum.Beside?
And what if that enum is evolving with new values. Wouldn't it be simpler and less verbose to convert the string to a number?
https://code.visualstudio.com/api/references/vscode-api#ViewColumn

Reply:
That's an interesting idea. Just change it to string, then map it back into the enum?

Reply:
If we change it to an optional string that corresponds to the vscode.ViewColumn enum value, we can then do things like `didact://?commandId=vscode.didact.startDidact&text=https://raw.githubusercontent.com/redhat-developer/vscode-didact/master/examples/copyFileURL.example.didact.md$$Beside`

Reply:
Or `didact://?commandId=vscode.didact.startDidact&text=https://raw.githubusercontent.com/redhat-developer/vscode-didact/master/examples/copyFileURL.example.didact.md$$Two`